subsys/bluetooth/mesh/light_ctrl_srv.c

```diff
@@ -562,6 +562,12 @@ static void timeout(struct k_work *work)
 	if (!is_enabled(srv)) {
 		if (srv->resume && atomic_test_and_clear_bit(&srv->flags, FLAG_RESUME_TIMER)) {
 			LOG_DBG("Resuming LC server");
+			if (IS_ENABLED(CONFIG_BT_MESH_SCENE_SRV)) {
```

Comment:
This will not work in all cases. Scene recall should abort currently ongoing things in all of the following cases:
1. Light LC is transitioning somewhere due to FSM actions (i.e. in FADE ON or PROLONG states, i.e. in active transition)
2. Light is transitioning due to some other event, such as, receiving Generic Move Set message with non-zero transition time (Light LC FSM switches off already when such transition starts).
3. Light is transitioning due to previous scene recall. 

This gets slightly complicated by the fact that Scene-recall should do nothing, if it is already in the scene that is being recalled.

This change only solves the issue with (1). The call to 'scene_invalidate' at the end of the timeout() is still incorrect and should be moved at the time of starting of transition triggered due to something other than scene-recall.

Reply:
Isn't this all covered in the scene_recall function?

Reply:
1. Aren't we in the same scene until Run state timeout? Isn't this considered as transition complete for the light controller?
2. For any message coming from an extended model, the extended model does the invalidation, for example: https://github.com/nrfconnect/sdk-nrf/blob/abb1738e08e66587d09801f9db9ff54ea51f6cd5/subsys/bluetooth/mesh/gen_lvl_srv.c#L73
3. This is responsibility of the scene recall server, isn't it? If new scene is recalled, the scene server will trigger scene recall.

Reply:
Also:
- Mode state change: https://github.com/nrfconnect/sdk-nrf/blob/abb1738e08e66587d09801f9db9ff54ea51f6cd5/subsys/bluetooth/mesh/light_ctrl_srv.c#L811
- Light OnOff state change: https://github.com/nrfconnect/sdk-nrf/blob/abb1738e08e66587d09801f9db9ff54ea51f6cd5/subsys/bluetooth/mesh/light_ctrl_srv.c#L898
- etc.

Reply:
Right, I think I forgot about this. Then this looks ok.